WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -45,12 +45,13 @@ final class OrderDetailsDataSource: NSObject {
     /// Whether the order is eligible for card present payment.
     ///
     var isEligibleForCardPresentPayment: Bool {
-        return isOrderAmountEligibleForCardPayment() &&
-            isOrderCurrencyEligibleForCardPayment() &&
-            isOrderStatusEligibleForCardPayment() &&
-            isOrderPaymentMethodEligibleForCardPayment() &&
-            hasCardPresentEligiblePaymentGatewayAccount() &&
-            !orderContainsAnySubscription()
+        let hasCardPresentEligiblePaymentGatewayAccount = resultsControllers
+            .paymentGatewayAccounts
+            .contains(where: \.isCardPresentEligible)
+        let orderIsEligibleForCardPresentPayment = order.isEligibleForCardPresentPayment(cardPresentPaymentsConfiguration: cardPresentPaymentsConfiguration,
+                                                                                         products: resultsControllers.products)
```

Comment:
@koke @joshheald Thanks a lot for your feedback here 👏 
Notice that `checkCreationEligibility` action is in Yosemite because it actually triggers a remote call to check the eligibility, so it is done in the backend. In this case, we do the calculation locally, so, given that this logic is related to this specific view I would lean towards leaving it in `WooCommerce`. I also would understand that we move this logic to `Yosemite` because it is very related to the model entity.
For now, I will go and merge this one, but we can reconsider this topic when applying it to the payment methods.